packages/visual-editor/src/types/treeActions.ts

```diff
@@ -0,0 +1,54 @@
+import { CompositionComponentNode } from '@contentful/experience-builder-core/types';
+
+export type TreeDiffType =
+  | 'remove_node'
+  | 'add_node'
+  | 'move_node'
+  | 'update_node'
+  | 'reorder_node'
+  | 'replace_node';
```

Comment:
Could be nice to switch this type to an Enum since each string is used in a few places

Reply:
Yeah good call, went ahead and did this